surfer/viz.py

```diff
@@ -2178,19 +2179,16 @@ def save_movie(self, fname, time_dilation=4., tmin=None, tmax=None,
             raise ValueError("tmin=%r is smaller than the first time point "
                              "(%r)" % (tmin, self._times[0]))
 
+        # find indexes at which to create frames
+        tstep = 1. / (framerate * time_dilation)
         if tmax is None:
             tmax = self._times[-1]
-        elif tmax >= self._times[-1]:
+        elif tmax > self._times[-1]:
             raise ValueError("tmax=%r is greater than the latest time point "
                              "(%r)" % (tmax, self._times[-1]))
-
-        # find indexes at which to create frames
-        tstep = 1. / (framerate * time_dilation)
-        if np.allclose((tmax - tmin) % tstep, 0):
-            tstop = tmax + tstep / 2.
-        else:
-            tstop = tmax
-        times = np.arange(tmin, tstop, tstep)
+        times = np.arange(tmin, tmax + sys.float_info.epsilon, tstep)
+        while times[-1] > tmax:
+            times = times[:-1]
```

Comment:
I could use `if`, I don't assume there are ever more than 1 values larger than tmax. `times = times[times < tmax]` seems less efficient because you create and copy arrays in every iteration